sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -653,35 +654,14 @@ size_t coap_put_option(uint8_t *buf, uint16_t lastonum, uint16_t onum, const uin
     return (size_t)n;
 }
 
-size_t coap_put_option_ct(uint8_t *buf, uint16_t lastonum, uint16_t content_type)
-{
-    if (content_type == 0) {
-        return coap_put_option(buf, lastonum, COAP_OPT_CONTENT_FORMAT, NULL, 0);
-    }
-    else if (content_type <= 255) {
-        uint8_t tmp = content_type;
-        return coap_put_option(buf, lastonum, COAP_OPT_CONTENT_FORMAT, &tmp, sizeof(tmp));
-    }
-    else {
-        return coap_put_option(buf, lastonum, COAP_OPT_CONTENT_FORMAT, (uint8_t *)&content_type, sizeof(content_type));
-    }
-}
-
 static unsigned _size2szx(size_t size)
 {
-    unsigned szx = 0;
-    assert(size <= 1024);
+    assert(size >= 16 && size <= 1024);
 
-    while (size) {
-        size = size >> 1;
-        szx++;
-    }
-    /* Size exponent + 1 */
-    assert(szx >= 5);
-    return szx - 5;
+    return bitarithm_lsb(size) - 4;
```

Comment:
I think I was too aggressive by returning here. I'd like to restore the assert on the szx value. I'll add in a fix.

Apologies for the churn.

Reply:
ok!

Reply:
Just pushed the fixup. Also added a comment to explain the sequencing of the offset subtraction.